src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1636 @@
+                        )
+            return icc_profile
+        else:
+            return None
+
+    def _add_icc_profile(
+        self,
+        referenced_images: Sequence[Dataset],
+        icc_profile: Optional[bytes] = None,
+    ) -> None:
+        """Add elements of ICC Profile module to the dataset.
+
+        Parameters
+        ----------
+        referenced_images: Sequence[pydicom.Dataset]
+            Images referenced in the presentation state.
+        icc_profile: Union[bytes, None], optional
+            ICC color profile to include in the presentation state. If none is
+            provided, the profile will be copied from the referenced images.
+            The profile must follow the constraints listed in :dcm:`C.11.15
+            <part03/sect_C.11.15.html>`.
+
+        """
+        if self.SOPClassUID in (
+            ColorSoftcopyPresentationStateStorage,
+            PseudoColorSoftcopyPresentationStateStorage
+        ):
+            if icc_profile is None:
+                # Copy profile from referenced images if none was provided
+                icc_profile = self._get_icc_profile(referenced_images[0])
+
+                # Ensure that there are no different profiles found in other
+                # referenced_images
+                if len(referenced_images) > 1:
+                    for ref_im in referenced_images[1:]:
+                        new_profile = self._get_icc_profile(ref_im)
+                        if new_profile != icc_profile:
+                            raise ValueError(
+                                'Non-matching ICC profiles were found in the '
+                                'referenced images. In this case, a profile '
+                                'must be manually specified via the '
+                                '"icc_profile" argument.'
+                            )
+            else:
+                # Check that the bytes represent a valid profile
+                reconstructed_profile = ImageCmsProfile(BytesIO(icc_profile))
+
+                if reconstructed_profile.profile.device_class != 'scnr':
+                    raise ValueError(
+                        'The device class of the ICC Profile must be "scnr", '
+                        f'got "{reconstructed_profile.profile.device_class}".'
+                    )
+                if reconstructed_profile.profile.xcolor_space != 'RGB':
+                    raise ValueError(
+                        'The color space of the ICC Profile must be "RGB", '
+                        f'got "{reconstructed_profile.profile.xcolor_space}".'
+                    )
+                pcs = reconstructed_profile.profile.connection_space
+                if pcs not in ('Lab', 'XYZ'):
+                    raise ValueError(
+                        'The profile connection space of the ICC Profile must '
+                        f'be "Lab" or "XYZ", got "{pcs}".'
+                    )
```

Comment:
```suggestion
                cms_profile = ImageCmsProfile(BytesIO(icc_profile))

                device_class = cms_profile.profile.device_class.strip()
                if device_class not in ('scnr', 'spac'):
                    raise ValueError(
                        'The device class of the ICC Profile must be "scnr" '
                        f'or "spac", got "{device_class}".'
                    )
                color_space = cms_profile.profile.xcolor_space.strip()
                if color_space != 'RGB':
                    raise ValueError(
                        'The color space of the ICC Profile must be "RGB", '
                        f'got "{color_space}".'
                    )
                pcs = cms_profile.profile.connection_space.strip()
                if pcs not in ('Lab', 'XYZ'):
                    raise ValueError(
                        'The profile connection space of the ICC Profile must '
                        f'be "Lab" or "XYZ", got "{pcs}".'
                    )
```

Reply:
That suggestion makes sure that the strings are formatted correctly (they may contain trailing white spaces). It also includes `"spac"` as a valid device class.